Makefile.package

```diff
@@ -304,15 +333,15 @@ ${SOURCE_DIR}:
 
 ${SOURCE_CMAKELISTS}: ${SOURCE_DIR}
 
-${BUILD_CONANINFO}: ${SOURCE_CONANFILE}
+${BUILD_CONANINFO}: ${SOURCE_CONANFILE} ${BUILD_LOCKFILE}
 	# Install package dependencies and prepare in-source build.
 	#
-	mkdir -p ${BUILD_DIR}
+	mkdir -p "${BUILD_DIR}"
 	conan install . ${PACKAGE_FQN} \
 		--install-folder=${BUILD_DIR} \
 		-s ${PACKAGE_NAME}:build_type=${BUILD_TYPE} \
```

Comment:
This does not work if you are using a lockfile. The `build_type`setting must be set during lockfile creation (as you did) and must not be set during build. 

Steps to reproduce:
```
make -C models clean all LOCKFILE_SOURCE=../conanfile.py

ERROR: Cannot use profile, settings, options, env or conf 'host' when using lockfile
```
- [ ] Fix obsolete `build_type` setting when lockfile is used

Reply:
Oof.